### PR TITLE
Add GPU Support for ONNX Models

### PR DESCRIPTION
**Pull request by KarimLulu**
This PR enables GPU acceleration for ONNX models in GLiNER by adding support for CUDA execution. 
The changes ensure that ONNX models can leverage GPU resources when available, leading to improved inference performance.

**Changes**

1. Updated Device Determination
Modified the device property to correctly determine the device for both PyTorch and ONNX models.
For ONNX models, the code now checks if `CUDAExecutionProvider` is in the execution providers and returns the appropriate device.

2. Unified Tensor Device Placement
Now, tensors are always moved to `self.device`, ensuring correct tensor placement and consistency between ONNX and PyTorch models.

3. Modified Model Loading
Adjusted the `_from_pretrained` method to set ONNX Runtime execution providers based on the map_location parameter.
If map_location is 'cuda', it sets the providers to `['CUDAExecutionProvider']`; otherwise, it defaults to `['CPUExecutionProvider']`.

4. Updated Dependencies
Added `onnxruntime-gpu` to `pyproject.toml` and `requirements.txt` to ensure GPU support

**Usage**
To enable GPU acceleration for ONNX models:
```
model = GLiNERModel.from_pretrained(
    'path_or_repo_id',
    load_onnx_model=True,
    onnx_model_file='model.onnx',
    map_location='cuda'  # Set to 'cpu' if GPU is not available
)
```

### CODE REVIEW

File: gliner/model.py [hunk 1]
```diff
@@ -831,7 +834,12 @@ def _from_pretrained(
                 session_options.graph_optimization_level = (
                     ort.GraphOptimizationLevel.ORT_ENABLE_ALL
                 )
-            ort_session = ort.InferenceSession(model_file, session_options)
+            providers = ['CPUExecutionProvider']
+            if map_location == 'cuda':
```
Comment (zoeypz) [inline gliner/model.py:838]: map_location could be "cuda:0" ?

Comment (Ingvarstep) [reply gliner/model.py:838]: Yeah, it will be more general to make it like that:
`if "cuda" in map_location`

**Comment by Ingvarstep** [comment]
@KarimLulu , great work in general. However, I am not sure that it correctly utilizes CUDAExecutionProvider, have you tested that actually uses it?

File: gliner/model.py (hunk 1 above)
Comment (KarimLulu) [reply gliner/model.py:838]: great catch, let me fix that

Comment (KarimLulu) [reply gliner/model.py]: done

**Comment by KarimLulu** [comment]
> @KarimLulu , great work in general. However, I am not sure that it correctly utilizes CUDAExecutionProvider, have you tested that actually uses it?

thanks @Ingvarstep for the review!
yeah, we are running this in prod now on 1 GPU T4 sagemaker instances, and we got 3x latency and throughput improvements compared to the plain CPU ONNX

**Comment by KarimLulu** [comment]
@Ingvarstep Please let me know if there are any further adjustments needed. Thanks!

**Comment by Ingvarstep** [comment]
@KarimLulu , thank you for your contribution, I think it satisfies all requirements.

**Pull request merged**